malariagen_data/pf7.py

```diff
@@ -0,0 +1,278 @@
+        data_vars["variant_is_snp"] = [DIM_VARIANT], is_snp
+
+        # variant_numalt
+        numalt_z = root["variants/numalt"]
+        numalt = da_from_zarr(numalt_z, inline_array=inline_array, chunks=chunks)
+        data_vars["variant_numalt"] = [DIM_VARIANT], numalt
+
+        # variant_CDS
+        cds_z = root["variants/CDS"]
+        cds = da_from_zarr(cds_z, inline_array=inline_array, chunks=chunks)
+        data_vars["variant_CDS"] = [DIM_VARIANT], cds
+
+        # call arrays
+        gt_z = root["calldata/GT"]
+        call_genotype = da_from_zarr(gt_z, inline_array=inline_array, chunks=chunks)
+        ad_z = root["calldata/AD"]
+        call_ad = da_from_zarr(ad_z, inline_array=inline_array, chunks=chunks)
+        data_vars["call_genotype"] = (
+            [DIM_VARIANT, DIM_SAMPLE, DIM_PLOIDY],
+            call_genotype,
+        )
+        data_vars["call_AD"] = ([DIM_VARIANT, DIM_SAMPLE, DIM_ALLELE], call_ad)
+
+        # sample arrays
+        z = root["samples"]
+        sample_id = da_from_zarr(z, inline_array=inline_array, chunks=chunks)
+        coords["sample_id"] = [DIM_SAMPLE], sample_id
+
+        if extended:
+            for var_name in self.extended_calldata_variables:
+                z = root[f"calldata/{var_name}"]
+                var = da_from_zarr(z, inline_array=inline_array, chunks=chunks)
+                data_vars[f"call_{var_name}"] = (
+                    self.extended_calldata_variables[var_name],
+                    var,
+                )
+
+            for var_name in self.extended_variant_fields:
+                field_z = root[f"variants/{var_name}"]
+                field = da_from_zarr(field_z, inline_array=inline_array, chunks=chunks)
+                data_vars[f"variant_{var_name}"] = (
+                    self.extended_variant_fields[var_name],
+                    field,
+                )
+
+        # create a dataset
+        ds = xarray.Dataset(data_vars=data_vars, coords=coords)
+
+        return ds
+
+    def variant_calls(self, extended=False, inline_array=True, chunks="native"):
+        """Access variant sites, site filters and genotype calls.
+        Parameters
+        ----------
+        inline_array : bool, optional
+            Passed through to dask.array.from_array().
+        chunks : str, optional
+            If 'auto' let dask decide chunk size. If 'native' use native zarr chunks.
+            Also can be a target size, e.g., '200 MiB'.
+        Returns
+        -------
+        ds : xarray.Dataset
+        """
+
+        datasets = [
+            self._variant_dataset(
+                extended=extended,
+                inline_array=inline_array,
+                chunks=chunks,
+            )
+        ]
+        ds = xarray.concat(
+            datasets,
+            dim=DIM_VARIANT,
+            data_vars="minimal",
+            coords="minimal",
+            compat="override",
+            join="override",
+        )
```

Comment:
There's no need to concatenate here, there's only ever one dataset.

Reply:
Sorry, I meant to do this but forgot. I have removed the concatenation now. I still have `variant_calls` as a separate function that calls `_variant_dataset` though, do you think it makes sense to rename `_variant_dataset` to `variant_calls` and just have one function or keep them separate in case we add any options to filter the dataset?

Reply:
I'd say just have one function, no point in having two functions if they're not actually doing anything different. Can always split out later if it serves some purpose.

Reply:
I have merged _variant_dataset and variant_calls into one function